kirin/tasks.py

```diff
@@ -96,95 +98,66 @@ def poller(self):
-        purge_trip_update.delay(config)
-
+            logger.info("executing '{}' for contributor '{}'".format(process_name, contributor.id))
 
-@celery.task(bind=True)
-def purge_rt_update_by_connector_type(self, connector_type):
-    """
-    This task will remove realtime update
-    """
-    for contributor in Contributor.find_by_connector_type(connector_type.value, include_deactivated=True):
-        config = {"contributor": contributor.id, "nb_days_to_keep": contributor.nb_days_to_keep_rt_update}
-        purge_rt_update.delay(config)
+            deleted = self.delete_rows(contributor)
+            logger.info("operation '{}' for contributor '{}' is finished".format(process_name, contributor.id))
+            return deleted
 
+    @abstractmethod
+    def delete_rows(self, contributor: Contributor) -> Dict:
+        raise NotImplementedError
 
-@celery.task(bind=True)
-def purge_gtfs_trip_update(self):
-    purge_trip_update_by_connector_type(ConnectorType.gtfs_rt)
 
+class RealTimeUpdatePurge(AbstractPurge):
+    def delete_rows(self, contributor: Contributor) -> Dict:
+        until = datetime.date.today() - datetime.timedelta(days=int(contributor.nb_days_to_keep_rt_update))
+        return RealTimeUpdate.remove_by_contributors_until(contributors=[contributor.id], until=until)
 
-@celery.task(bind=True)
-def purge_gtfs_rt_update(self):
-    purge_rt_update_by_connector_type(ConnectorType.gtfs_rt)
 
+class TripUpdatePurge(AbstractPurge):
+    def delete_rows(self, contributor: Contributor) -> Dict:
+        until = datetime.date.today() - datetime.timedelta(days=int(contributor.nb_days_to_keep_trip_update))
+        return TripUpdate.remove_by_contributors_until(contributors=[contributor.id], until=until)
 
-@celery.task(bind=True)
-def purge_piv_trip_update(self):
-    purge_trip_update_by_connector_type(ConnectorType.piv)
 
-
-@celery.task(bind=True)
-def purge_piv_rt_update(self):
-    purge_rt_update_by_connector_type(ConnectorType.piv)
-
-
-@celery.task(bind=True)
-def purge_cots_trip_update(self):
-    purge_trip_update_by_connector_type(ConnectorType.cots)
+def _merge_deleted(original, append):
+    for key, value in append.items():
+        if key in original:
+            original[key] = original[key] + value
+        else:
+            original[key] = value
 
 
 @celery.task(bind=True)
-def purge_cots_rt_update(self):
-    purge_rt_update_by_connector_type(ConnectorType.cots)
+def purge(self):
+    contributors = Contributor.query.all()
+    tu_purge = TripUpdatePurge()
+    rtu_purge = RealTimeUpdatePurge()
+    deleted = {}
+    for contributor in contributors:
+        new_deleted = tu_purge.do_purge(contributor)
+        _merge_deleted(deleted, new_deleted)
+        new_deleted = rtu_purge.do_purge(contributor)
+        _merge_deleted(deleted, new_deleted)
+    new_relic.record_custom_parameter("nb_rtu_purged", deleted[RealTimeUpdate])
```

Comment:
To match other metrics vocabulary (output_trip_update_count)
```suggestion
    new_relic.record_custom_parameter("total_rtu_purged_count", deleted[RealTimeUpdate])
```
We may even have `real_time_update` instead of `rtu`.

Reply:
I've done this aba0f1d, tell me what you think about it.

Reply:
And even this 2dddf16 to make the name longer.